src/js/components/Select/SelectContainer.js

```diff
@@ -218,22 +210,111 @@ class SelectContainer extends Component {
     }
   };
 
+  optionLabel = index => {
+    const { options, labelKey } = this.props;
+    const option = options[index];
+    let optionLabel;
+    if (labelKey) {
+      if (typeof labelKey === 'function') {
+        optionLabel = labelKey(option);
+      } else {
+        optionLabel = option[labelKey];
+      }
+    } else {
+      optionLabel = option;
+    }
+    return optionLabel;
+  };
+
+  optionValue = index => {
+    const { options, valueKey } = this.props;
+    const option = options[index];
+    let optionValue;
+    if (valueKey) {
+      if (typeof valueKey === 'function') {
+        optionValue = valueKey(option);
+      } else {
+        optionValue = option[valueKey];
+      }
+    } else {
+      optionValue = option;
+    }
+    return optionValue;
+  };
+
+  isDisabled = index => {
+    const { disabled, disabledKey, options, valueKey } = this.props;
+    const option = options[index];
+    let result;
+    if (!disabled) {
+      result = false;
+    } else if (disabledKey) {
+      if (typeof disabledKey === 'function') {
+        result = disabledKey(option, index);
+      } else {
+        result = option[disabledKey];
+      }
+    } else if (typeof disabled === 'function') {
+      result = disabled(option);
+    } else {
+      let value;
+      if (valueKey) {
+        if (typeof valueKey === 'function') {
+          value = valueKey(option);
+        } else {
+          value = option[valueKey];
+        }
+      } else {
+        value = option;
+      }
+      result = disabled.indexOf(value) !== -1;
+    }
+    return result;
+  };
+
+  isSelected = index => {
+    const { selected, value, valueKey } = this.props;
+    let result;
+    if (selected) {
+      // deprecated in favor of value
+      result = selected.indexOf(index) !== -1;
+    } else {
+      const optionValue = this.optionValue(index);
+      if (Array.isArray(value)) {
+        if (value.length === 0) {
+          result = false;
+        } else if (typeof value[0] !== 'object') {
+          result = value.indexOf(optionValue) !== -1;
+        } else if (valueKey) {
+          result = value.some(valueItem => {
```

Comment:
I'd love to see some maps created to enable O(1) access for `selected`, `disabled` and `values` (and anything else that may normally need to be looped over) to benefit a Select with 1000+ items. It is certainly more work to accomplish this and to ensure the maps are always being updated appropriately where necessary, but it guarantees speedy performance rather than doing `some`, `forEach`, `map` over all items on each render. This is less of an issue when infinite scroll changes to only render what's visible, but ultimately the most performant option regardless of implementation.